python/requirements_ml_docker.txt

```diff
@@ -4,14 +4,11 @@ ipython
 tblib
 
 # If you make changes to the torch versions, please also make the corresponding changes to `requirements_dl.txt`!
--f https://download.pytorch.org/whl/torch_stable.html
-torch==1.9.0+cu111
-torchvision==0.10.0+cu111
 
--f https://data.pyg.org/whl/torch-1.9.0+cu111.html
+--extra-index-url https://download.pytorch.org/whl/cu111  # for GPU versions of torch, torchvision
+--find-links https://data.pyg.org/whl/torch-1.9.0+cu111.html  # for GPU versions of torch-scatter, torch-sparse
+torch==1.9.0
+torchvision==0.10.0
 torch-scatter==2.0.9
 torch-sparse==0.6.12
-
-# torch-geometric has to be installed after torch-scatter and torch-sparse.
-torch-geometric==2.0.3; python_version < '3.7'
-torch-geometric==2.0.4; python_version >= '3.7'
+torch-geometric~=2.0.3
```

Comment:
@amogkam this will get 2.0.3 on py36 and 2.0.* on py37+

the order of statements in a reqs.txt is ignored by pip btw

Reply:
if cpu ml images come back ref https://github.com/ray-project/ray/issues/22468, this will have to be duplicated in `python/requirements/ml/requirements_dl.txt` ref https://github.com/ray-project/ray/pull/23580/files#r846430573

Reply:
Thanks @ddelange! Can we hard pin torch-geometric (even if we need separate lines for python 3.6 vs. 3.7? This will prevent possible breakages if there is an incompatible torch geometric release.

Reply:
And yes we did add the CPU ML image back in Ray 1.12.1. Like you said, could we add these lines to requirements_dl.txt? Thanks!

Reply:
sure :)

```suggestion
torch-geometric==2.0.3; python_version < '3.7'
torch-geometric==2.0.4; python_version >= '3.7'
```